app/src/main/java/com/github/kulebin/myfoursquareapp/App.java

```diff
@@ -0,0 +1,28 @@
+package com.github.kulebin.myfoursquareapp;
+
+import android.app.Application;
+
+import com.github.kulebin.myfoursquareapp.thread.IThreadManager;
+
+public class App extends Application {
+
+    private IThreadManager mThreadManager;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        ContextHolder.set(this);
+    }
+
+    @Override
+    public Object getSystemService(final String pName) {
+        if (IThreadManager.APP_SERVICE_KEY.equals(pName)) {
+            if (mThreadManager == null) {
```

Comment:
Such lazy initialization is good for classes we know user can't use at all for a session. 
Threading, networking, etc. are always used, please create instance in onCreate